### ISSUE TITLE
Errors installing Emscripten during setup steps on a Windows machine

### ISSUE THREAD

**Issue by chrispcampbell** [climateinteractive/SDEverywhere#359]
### Discussed in https://github.com/climateinteractive/SDEverywhere/discussions/358

<div type='discussions-op-text'>

<sup>Originally posted by **andyhe958** September 25, 2023</sup>
Hi everyone,

I have been trying to run SDEverywhere from command prompt on a Windows machine but came across an error below when installing Emscripten. It suggests Python is not found, but in fact I do have Python 3.11 installed.

![Error message 1](https://github.com/climateinteractive/SDEverywhere/assets/146044517/2c6aea19-0e85-4544-bd01-75001671e94a)

Then I went on to disable the app alias for Python as suggested, but I got a second error, which seems to contradict with the first error:

![Error message 2](https://github.com/climateinteractive/SDEverywhere/assets/146044517/58217792-05c1-4088-981b-b7ff357827ee)

I have since managed to install Emscripten separately, not as part of the SDEverywhere setup steps, and it was installed successfully. I think I would post this here to help others potentially in a similar situation, and to help with further configuration of this tool for Windows.</div>

**Comment by tom-papaioannou** [comment]
> ### Discussed in #358
> 
> Originally posted by **andyhe958** September 25, 2023 Hi everyone,
> 
> I have been trying to run SDEverywhere from command prompt on a Windows machine but came across an error below when installing Emscripten. It suggests Python is not found, but in fact I do have Python 3.11 installed.
> 
> ![Error message 1](https://private-user-images.githubusercontent.com/146044517/270509481-2c6aea19-0e85-4544-bd01-75001671e94a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTE5NzE5OTUsIm5iZiI6MTcxMTk3MTY5NSwicGF0aCI6Ii8xNDYwNDQ1MTcvMjcwNTA5NDgxLTJjNmFlYTE5LTBlODUtNDU0NC1iZDAxLTc1MDAxNjcxZTk0YS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNDAxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDQwMVQxMTQxMzVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iZjEyYmJlMzhmNGM2NWZmM2Y4NjMwZDdlNjk4YjFmMDUzM2M4Zjk2OTE3MTQyZmQyNmU4MWQwY2JjODU1M2UyJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.HidyhtRokw7ZtzefL4q0x-vmssDiZmIwpQY1H9RhQDk)
> 
> Then I went on to disable the app alias for Python as suggested, but I got a second error, which seems to contradict with the first error:
> 
> ![Error message 2](https://private-user-images.githubusercontent.com/146044517/270509549-58217792-05c1-4088-981b-b7ff357827ee.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTE5NzE5OTUsIm5iZiI6MTcxMTk3MTY5NSwicGF0aCI6Ii8xNDYwNDQ1MTcvMjcwNTA5NTQ5LTU4MjE3NzkyLTA1YzEtNDA4OC05ODFiLWI3ZmYzNTc4MjdlZS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNDAxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDQwMVQxMTQxMzVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0wNGRlYmE2Y2RlN2JlZDJjNWE4MzY3MDlhNGZjMDgzZDQ5ZTVkZTc1NDI3Njg5YjU1ZmQzNDZiODQ4MWUxYjVmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.zbyrR1B-uUv9ZUXiYoiO9Vpb4Ek4EfBvH0M6ildqFec)
> 
> I have since managed to install Emscripten separately, not as part of the SDEverywhere setup steps, and it was installed successfully. I think I would post this here to help others potentially in a similar situation, and to help with further configuration of this tool for Windows.

Hello,

I had the same problem using git-bash environment.
I have edited the bashrc with nano

nano ~/.bashrc

to edit it and add at the end of the file this line to add it as alias
alias python3='python'

and saved. After a restart of the git-bash environment it worked.

If you use a windows terminal as CMD, I would create a python3.bat file and edit it to add these lines

@echo off
python %*

then procceed to add the folder of the bat file at the environment PATH variables in Windows settings.

**Comment by chrispcampbell** [comment]
Hi @MSDRL, it looks like from the first screenshots that you have installed Emscripten  separately yourself using Chocolatey, is that correct?  If so, then when you run the `npm create` command, you should select “no” (i.e. type “n”) when it asks if you would like to install the Emscripten SDK.  In other words, if you already have Emscripten installed, then there’s no point in having the `create` script do it for you, and you should skip that step (thus avoiding these Windows errors).

Note that we have recently added support for generating pure JavaScript code by default in part to avoid these issues you’re facing with Emscripten on Windows.  New users who simply want to evaluate SDEverywhere should not have to worry about these steps and can get up and running faster when Wasm is not involved.  These changes have not yet been “npm published” but will be published the week of July 8.

To summarize, you can either try my suggestion above about skipping the step, or wait a couple weeks at which point these install issues should go away.

**Comment by chrispcampbell** [comment]
@MSDRL: To follow up on my last comments, the changes I was referring to have been published, so now if you follow the [Quick Start](https://github.com/climateinteractive/SDEverywhere?tab=readme-ov-file#quick-start) instructions, when it asks you if you want to use WebAssembly, select "no" (the default) and it will create a project that uses JavaScript only (no Emscripten or generation of WebAssembly required).

See the [August 2024 Updates](https://github.com/climateinteractive/SDEverywhere/discussions/513) announcement for more details.

I still plan to smooth out this issue with Emscripten installation on Windows, but for most projects, the generated JavaScript code will be sufficient to get started, and compiling to WebAssembly can always be configured later if needed for faster model execution.
